src/bp-blogs/bp-blogs-template.php

```diff
@@ -373,13 +373,17 @@ function bp_get_blog_avatar( $args = '' ) {
 		// Set default value for the `alt` attribute.
 		$alt_attribute = __( 'Site icon for the blog', 'buddypress' );
 
-		if ( ! $blogs_template && isset( $args['blog_id'] ) && $args['blog_id'] ) {
+		$has_current_blog = isset( $blogs_template ) && isset( $blogs_template->blog->blog_id );
+
+		if ( false === $has_current_blog && isset( $args['blog_id'] ) && $args['blog_id'] ) {
```

Comment:
IMO, we should respect if the `blog_id` is passed, regardless if there is a blog in the current loop. But, I assume this logic is here for a reason, so let's keep it.

But technically, **it's not possible** to pass another, different `blog_id` into `bp_get_blog_avatar` inside a blog loop because of this logic.

Reply:
I agree with you, maybe we can fix it later in another ticket?

Reply:
Created: https://buddypress.trac.wordpress.org/ticket/9228